projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/IpTest.java

```diff
@@ -0,0 +1,21 @@
+package org.batfish.datamodel;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests of {@link Ip}. */
+@RunWith(JUnit4.class)
+public class IpTest {
+  @Test
+  public void numSubnetBitsToSubnetLong() throws Exception {
```

Comment:
I tested the first version of this for all numbers 0-31 against the existing implementation.